blackjax/kernels.py

```diff
@@ -785,47 +880,67 @@ def meads(
 
     """
 
-    kernel = ghmc.kernel(divergence_threshold=divergence_threshold)
+    step_fn = ghmc.kernel()
+
+    init, update = adaptation.meads.base(
+        logprob_grad_fn or jax.grad(logprob_fn),
+    )
+
+    batch_init = batch_fn(lambda r, p: ghmc.init(r, p, logprob_fn, logprob_grad_fn))
 
-    def kernel_factory(step_size: PyTree, alpha: float, delta: float):
-        def kernel_fn(rng_key, state):
-            return kernel(
+    def one_step(carry, rng_key):
+        states, adaptation_state = carry
+
+        def kernel(rng_key, state):
```

Comment:
How about
```suggestion
        def kernel_closure(rng_key, state):
```

Reply:
This one is truly a kernel in the sense of the original design document. However I generally agree with you that there's a naming issue across the library and we seriously need to work on it see #280 for instance.